string.go

```diff
@@ -501,6 +501,68 @@ func execAppend(db *DB, args [][]byte) redis.Reply {
 	return reply.MakeIntReply(int64(len(bytes)))
 }
 
+// execSetRange overwrites part of the string stored at key, starting at the specified offset.
+// If the offset is larger than the current length of the string at key, the string is padded with zero-bytes.
+func execSetRange(db *DB, args [][]byte) redis.Reply {
+	key := string(args[0])
+	offset, errNative := strconv.ParseInt(string(args[1]), 10, 64)
+	if errNative != nil {
+		return reply.MakeErrReply(errNative.Error())
+	}
+	value := args[2]
+	bytes, err := db.getAsString(key)
+	if err != nil {
+		return err
+	}
+	bytesLen := int64(len(bytes))
+	if bytesLen < offset {
+		diff := offset - bytesLen
+		diffArray := make([]byte, diff, diff)
+		bytes = append(bytes, diffArray...)
+		bytesLen = int64(len(bytes))
+	}
+	for i := 0; i < len(value); i++ {
+		idx := offset + int64(i)
+		if idx >= bytesLen {
+			bytes = append(bytes, value[i])
+		} else {
+			bytes[idx] = value[i]
+		}
+	}
+	db.PutEntity(key, &DataEntity{
+		Data: bytes,
+	})
+	db.AddAof(makeAofCmd("setRange", args))
+	return reply.MakeBulkReply(bytes)
```

Comment:
SetRange returns the length of the string after it was modified by the command. see https://redis.io/commands/setrange